### PR TITLE
feat: switch to yeoman-env

### PR DESCRIPTION
**Pull request by jannyHou**
### Description

The follow up PR of https://github.com/strongloop/generator-loopback/pull/363, please make sure the `generator-loopback` PR get merged first and a new major version of it is released.

#### Related issues
https://github.com/strongloop/generator-loopback/issues/355

### Checklist

- [ ] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style
  guide](http://loopback.io/doc/en/contrib/style-guide.html)

### CODE REVIEW

File: bin/loopback-cli.js [hunk 1]
```diff
@@ -48,41 +48,39 @@ const cwd = process.cwd();
...
-
-// `yo` is adding flags converted to CamelCase
-const options = camelCaseKeys(opts, {exclude: ['--', /^\w$/, 'argv']});
-Object.assign(options, opts);
-
-const generator = env.create(command, {args});
-
-debug('env.run %j %j', args, options);
-env.run(args, options);
+env.lookup(function() {
```
[the remark this answers is not in the document]
Comment (jannyHou) [reply bin/loopback-cli.js]: Good point, while the implementation of `lookup(cb)` always returns `cb(null)`, see [code](https://github.com/yeoman/environment/blob/master/lib/resolver.js#L51). It looks like a problem of yeoman-environment, it doesn't propagate the `error` got from `this._tryRegistering`.

Anyway I am adding the `err` check :)

**Comment by jannyHou** [comment]
@bajtos Thanks for the review and suggestion!

> Please run the CLI manually to check that we are correctly setting up options and arguments, there have been issues with this integration point in the past.

Sure 👍The PR is written when I invoke the generator by `lb *`. 
And it's good to know the use cases of the manual verifications.

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "loopback-cli",
-  "version": "4.2.0",
+  "version": "5.0.0",
```
Comment (virkt25) [inline package.json]: Is this to prevent tests breaking downstream?

File: bin/loopback-cli.js [hunk 3]
```diff
@@ -36,10 +36,10 @@ process.env.SLC_COMMAND = 'loopback-cli';
 // therefore I am intentionally loading it only after we have
 // handled the "--version" case which becomes much faster as the result.
 const lbGenerator = require('generator-loopback');
-const yeoman = lbGenerator._yeoman; // generator-loopback should export _yeoman
-assert(yeoman, 'generator-loopback should export _yeoman');
+const yeomanEnv = lbGenerator._yeomanEnv; // generator-loopback should export _yeomanEnv
```
Comment (virkt25) [inline bin/loopback-cli.js]: nitpick: I don't think this comment is needed considering the assert on the next line

File: package.json (hunk 2 above)
Comment (jannyHou) [reply package.json]: sorry for replying late 🙈 

> prevent tests breaking downstream

Exactly.

**Comment by jannyHou** [comment]
FYI, the jenkins machines seem don't install the latest `generator-loopback`. They all fail with the one test that's been fixed in `generator-loopback@6.0.1`

Travis tests all pass.

I would like to merge this PR then create another PR to fix the jenkins error.

**Comment by b-admike** [comment]
> FYI, the jenkins machines seem don't install the latest `generator-loopback`. They all fail with the one test that's been fixed in `generator-loopback@6.0.1`
> 
> Travis tests all pass.
> 
> I would like to merge this PR then create another PR to fix the jenkins error.

Perhaps @rmg would know better / how to remove it if npm used a cached version of the module.

**Pull request merged**